plugins/fritz-box7490/de/fritzbox7490.sh

```diff
@@ -0,0 +1,204 @@
+#!/bin/bash
+# FritzBox.sh
+
+### Quelle des Codekerns - Target of die skript core ############################
+# https://github.com/Tscherno/Fritzbox.sh                                       #
+# compatible with Fritz.box Firmware 6.50 and higher, Horst Schmid, 2016-01-05  #
+# /usr/local/addons/cuxd/user/FritzBox.sh                                       # 
+#################################################################################
+
+#####################################################################################################################################################
+# Der Vorbereitungs-Job muss per cron gestartet werden - Thie prerunning job must be started by cron                                                #
+# /etc/cron.d/fritzbox                                                                                                                              #
+# 3,8,13,18,23,28,33,38,43,48,53,58 * * * * root /<patch to the job>/fritzbox7490.sh  -p <login passwd> (-u >login user if required>] #okay         #
+#                                                                                                                                                   #
+#                                                                                                                             Jo Hartmann V17.0126  #
+#####################################################################################################################################################
+
+# Deklaration der Dateipfade  
+# Declaration of file pathes
+  CPWMD5="/etc/munin/plugins/pre2run/cpwmd5"
+  FRITZLOGIN="/login_sid.lua"
+  FRITZWEBCM="/cgi-bin/webcm"
+  FRITZHOME="/home/home.lua"
+  CURLFILE="/var/tmp/FritzBoxCurl.txt"
+  LOGFILE="/var/log/fritzbox7490.log"
+  PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+# Wie werden die Webseiten aufgerufen
+# How to call web pages
+  WEBCLIENT="curl -s"
+
+# Deklaration sonstiger Variablen
+# Declaration of other variables
+  FritzBoxURL="192.168.0.111"
+  Username=""
+  Passwd=""
+  Debug=false
+
+# Verarbeitung der Aufruf-Parameter
+# Processing the script parameters
+  while getopts u:p:d OPT; do
+    case $OPT in
+       u)
+         Username=$OPTARG
+         ;;
+       p)
+         Passwd=$OPTARG
+         ;;
+       d)
+         Debug=true
+         LOGFILE="/dev/stdout"
+         ;;
+ 
+    esac
+  done
+
+# Anmelde-Routine, Quelle: https://github.com/Tscherno/Fritzbox.sh, Horst Schmid, 2016-01-05
+# Login routine, source:   https://github.com/Tscherno/Fritzbox.sh, Horst Schmid, 2016-01-05
+  LOGIN(){
```

Comment:
This function seems to do execute the login and additionally write loads of data to the logfile.
I strongly suggest decoupling both actions and move the redirections to the function call:
`LOGIN >>"$LOGFILE"`
This should make this function a bit more readable.
Additionally: please document its side-effects (assigned variables and so on, if there are any).